pkg/client/request.go

```diff
@@ -478,6 +483,27 @@ func (r *Request) Do() Result {
 			continue
 		}
 
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			if retries == 10 {
+				// Give up after about 10 failed attempts.
+				return Result{respBody, created, err, r.codec}
```

Comment:
This line of code appears 4 times. Could this be reorganized to fall through to one instance of it?